docs/JSON-RPC-API-using-jmwalletd.md

```diff
@@ -0,0 +1,118 @@
+
+* HTTP Request type: POST
+* Route: `/wallet/create`
+* POST body contents: {"walletname": walletname, "password": password, "wallettype": wallettype}
+* Returns: on success, {"walletname": walletname, "already_loaded": False}
+
+(TODO some confusion over two different walletnames here, I need to check, but the wallet name sent by the caller will be used for the file name, I believe).
+
+#### `unlockwallet`
+
+Open an existing wallet using a password.
+
+* HTTP Request type: POST
+* Route: `/wallet/<string:walletname>/unlock`
+* POST body contents: {"password": password}
+* Returns: on success, {"walletname": walletname, "already_loaded": True}
+
+(see previous on walletname, same applies here).
+
+#### `lockwallet`
+
+Stops the wallet service for the current wallet; meaning it cannot then be accessed without re-authentication.
+
+* HTTP Request type: GET
+* Route: `/wallet/<string:walletname>/lock`
+* Returns: on success, {"walletname": walletname}
+
+(see previous on walletname, same applies here).
+
+#### `displaywallet`
+
+Get JSON representation of wallet contents for wallet named `walletname`:
+
+* HTTP Request type: GET
+* Route: `/wallet/<string:walletname>/display`
+* Returns: a JSON object which is the entire wallet contents, mixdepth by mixdepth.
+    - Example output from a signet wallet is given at the bottom of the document.
+
+#### `maker/start`
+
+Starts the yield generator/maker service for the given wallet, using the IRC and tor network connections
+in the backend (inside the process started with jmwalletd).
+See Joinmarket yield generator config defaults in `jmclient.configure` module for info on the data that must
+be specified in the POST body contents.
+
+* HTTP Request type: POST
+* Route: `/wallet/<string:walletname>/maker/start`
+* POST body contents: {"txfee", "cjfee_a", "cjfee_r", "ordertype", "minsize"]
+* Returns: on success, {"walletname": walletname}
+
+(see previous on walletname, same applies here).
+
+#### `maker/stop`
+
+Stops the yieldgenerator/maker service if currently running for the given wallet.
+
+* HTTP Request type: GET
+* Route: `/wallet/<string:walletname>/maker/start`
+* Returns: on success, {"walletname": walletname}
+
+(see previous on walletname, same applies here).
+
+#### `snicker/start`
+
+Starts the SNICKER service (see [here](SNICKER.md)) for the given wallet. Note that this requires
+no configuration for now, though that is likely to change. Also note this is not yet supported for
+mainnet.
+
+* HTTP Request type: GET
+* Route: `/wallet/<string:walletname>/snicker/start`
+* Returns: on success, {"walletname": walletname}
+
+(see previous on walletname, same applies here).
+
+#### `snicker/stop`
+
+Stops the snicker service if currently running for the given wallet.
+
+* HTTP Request type: GET
+* Route: `/wallet/<string:walletname>/snicker/start`
+* Returns: on success, {"walletname": walletname}
+
```

Comment:
Similar to https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/996/files#r700534364 it'd be a good idea to add the new routes into this list.